scalding-beam/src/test/scala/com/twitter/scalding/beam_backend/BeamBackendTests.scala

```diff
@@ -35,6 +42,63 @@ class BeamBackendTests extends FunSuite with BeforeAndAfter {
   def tmpPath(suffix: String): String =
     Paths.get(testPath, suffix).toString
 
+  test("BeamOp caching: FromIterable") {
+    val a = TypedPipe.from(0 to 5)
+
+    val (pipeline, op) = beamPlan(a)
+
+    assert(op.isInstanceOf[FromIterable[Int]])
+    assert(op.run(pipeline) == op.run(pipeline))
```

Comment:
do you want `eq` here? to test reference equality?

Reply:
yes my bad

Reply:
done